lighthouse-docker/server.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+'use strict';
+
+const fs = require('fs');
+const express = require('express');
+const spawn = require('child_process').spawn;
+const bodyParser = require('body-parser');
+
+const PORT = process.env.PORT || 8080;
+
+// Handler for CI.
+function runLH(params, req, res, next) {
+  const url = params.url;
+  const format = params.output || params.format || 'html';
+  const log = req.method === 'GET' && !('nolog' in params);
+
+  if (!url) {
+    res.status(400).send('Please provide a URL.');
+    return;
+  }
+
+  const fileName = `report.${Date.now()}.${format}`;
+  const outputPath = `./home/chrome/reports/${fileName}`;
+
+  const args = [
+    `--output-path=${outputPath}`,
+    `--output=${format}`,
+    '--port=9222',
+    // Note: this is a noop when using Dockerfile.headful b/c Chrome is already
+    // launched when the container starts up.
+    `--chrome-flags="--headless"`,
+  ];
+  const child = spawn('lighthouse', [...args, url]);
```

Comment:
a little amusing that the server uses the CLI rather than node module, good example for folks to not try to run the node module in same process as a server :D

Reply:
Ha! TBH it was slightly easier when I put this together back in the day. The module didn't have a way to output the html report. Now we have `ReportGenerator` where one can use it to craft the HTML report from LHR. I'd say we should change it, but this seems to have decent perf and has been working for a while now.